map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,218 @@
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+
+import org.apache.commons.io.IOUtils;
+
+
+public class ShowLocationHistorySample extends Application {
+
+  private MapView mapView;
+  private boolean isTrackingEnabled = false;
+  private Point position;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Show Location History Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map with a dark gray canvas basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createDarkGrayCanvasVector());
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set the map views's viewpoint centered on Los Angeles, California and scaled
+      mapView.setViewpoint(new Viewpoint(new Point(-13185535.98, 4037766.28,
+        SpatialReferences.getWebMercator()), 7000));
+
+      // create a button that toggles the location tracking
+      Button trackingButton = new Button("Start tracking");
+      trackingButton.setDisable(true);
+
+      // create a graphics overlay for the points and use a red circle for the symbols
+      GraphicsOverlay locationHistoryOverlay = new GraphicsOverlay();
+      SimpleMarkerSymbol locationSymbol = new SimpleMarkerSymbol(
+        SimpleMarkerSymbol.Style.CIRCLE, 0xFFFF0000, 10f);
+      SimpleRenderer locationHistoryRenderer = new SimpleRenderer(locationSymbol);
+      locationHistoryOverlay.setRenderer(locationHistoryRenderer);
+
+      // create a graphics overlay for the lines connecting the points and use a green line for the symbol
+      GraphicsOverlay locationHistoryLineOverlay = new GraphicsOverlay();
+      SimpleLineSymbol locationLineSymbol = new SimpleLineSymbol(
+        SimpleLineSymbol.Style.SOLID, 0xFF00FF00, 2.0f);
+      SimpleRenderer locationHistoryLineRenderer = new SimpleRenderer(locationLineSymbol);
+      locationHistoryLineOverlay.setRenderer(locationHistoryLineRenderer);
+
+      // add the graphics overlays to the map view
+      mapView.getGraphicsOverlays().addAll(Arrays.asList(locationHistoryOverlay, locationHistoryLineOverlay));
+
+      // create a polyline builder to connect the location points
+      PolylineBuilder polylineBuilder = new PolylineBuilder(SpatialReferences.getWebMercator());
+
+      try {
+        // access the json of the location points
+        String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+          StandardCharsets.UTF_8);
+        // create a polyline from the location points
+        Polyline routePolyline = (Polyline) Geometry.fromJson(polylineData,
+          SpatialReferences.getWebMercator());
```

Comment:
this can all go on the one line without exceeding the recommended line limit 
```suggestion
        Polyline routePolyline = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWebMercator());
```